src/cbc_sdk/workload/sensor_lifecycle.py

```diff
@@ -0,0 +1,210 @@
+        self._full_init = True
+
+    @classmethod
+    def create(cls, cb, device_type, architecture, sensor_type, version):
+        """
+        Helper method used to create a SensorType from its four components.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            device_type (str): Device type to be used.  Valid values are "WINDOWS", "LINUX", and "MAC".
+            architecture (str): Architecture to be used.  Valid values are "32", "64", and "OTHER".
+            sensor_type (str): Sensor type to be used.  Valid values are "WINDOWS", "MAC", "RHEL", "UBUNTU", "SUSE",
+                               and "AMAZON_LINUX".
+            version (str): Sensor version number to be used.
+
+        Returns:
+            SensorType: A SensorType object with those specified values.
+
+        Raises:
+            ApiError: If an invalid value was used for one of the three limited values.
+        """
+        if device_type not in SensorType.VALID_DEVICE_TYPES:
+            raise ApiError("invalid device_type specified for SensorType")
+        if architecture not in SensorType.VALID_ARCHITECTURES:
+            raise ApiError("invalid architecture specified for SensorType")
+        if sensor_type not in SensorType.VALID_TYPES:
+            raise ApiError("invalid type specified for SensorType")
+        return SensorType(cb, {'device_type': device_type, 'architecture': architecture, 'sensor_type': sensor_type,
+                               'version': version})
+
+    def _as_dict(self):
+        """
+        Returns a dictionary containing the SensorType values as they are to be submitted to the server.
+
+        Returns:
+            dict: A dictionary containing the SensorType values.
+        """
+        return {'device_type': self.device_type, 'architecture': self.architecture, 'type': self.sensor_type,
+                'version': self.version}
+
+
+class SensorInfo(UnrefreshableModel):
+    """Represents information related tot he installation of a sensor."""
+    swagger_meta_file = "workload/models/sensorInfo.yaml"
+
+    def __init__(self, cb, initial_data=None):
+        """
+        Initialize the SensorInfo object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            initial_data (dict): Initial data used to populate the sensor information.
+        """
+        super(SensorInfo, self).__init__(cb, None, initial_data)
+        self._full_init = True
+        self._sensor_type = None
+
+    @property
+    def sensor_type_(self):
+        """
+        Returns the sensor type information in this information object, as a SensorType object.
+
+        Returns:
+            SensorType: The sensor type information.
+        """
+        if self._sensor_type is None:
+            self._sensor_type = SensorType(self._cb, self._info['sensor_type'])
+        return self._sensor_type
+
+
+class SensorRequest:
```

Comment:
This implementation is confusing where the two aspects are blended together and the SensorType model is not a model in the API it's a set of sub properties within the sensor model. My thoughts are that you should have a SensorKit model which combines the SensorType and SensorInfo along with a SensorKitQuery class similar to how the rest of the cbc sdk works.

`get_config_template` can be a classmethod on the SensorKit model

`add_sensor_type`, `expire_time`, and `config_params` would be functions on the SensorKitQuery then `get_sensor_info` would be replace by a `submit` function instead

`request_sensor_install` should be into the compute_resource model that Hristo created and then a class method could be used to support it in bulk. That way you could support a use case for the cbc sdk where a developer can run a function on the compute resource model `compute_resource.install_sensor(sensor_type, config_file=None)` or do it in bulk `ComputeResource.bulk_install(compute_resources, sensor_types, config_file=None)` Since the config file is optional in the request it can be optional in the function.


The above implementation would match closer to the rest of the cbc sdk and would integrate with the other models in the workloads package

Reply:
To do anything about this, I need two things.  First, I need Hristo's compute_resource model, which isn't in the code yet, but will be soon.  Second, I need you to explain more about this organization of functionality ("SensorKit"? Where did _that_ name come from?)  Until then, I'm at an impasse.

Reply:
Please look at the new organization and see if it matches up with what you were saying.